monitor/api/tests/tests.py

```diff
@@ -0,0 +1,163 @@
+import json
+from typing import Any, Dict, List, Optional
+
+from clients.api.api import APIClient
+from monitor.api.tests.schema import InvocationsSchems, TestMetadataSchema, TestUniqueIdType, InvocationSchema, TotalsInvocationsSchema
+
+
+class TestsAPI(APIClient):
+    @staticmethod
+    def _get_test_sub_type_unique_id(test: TestMetadataSchema) -> str:
+        return f"{test.test_unique_id}.{test.test_type}.{test.test_sub_type if test.test_sub_type else 'None'}.{test.table_name if test.table_name else 'None'}.{test.column_name if test.column_name else 'None'}"
+
+    def get_tests_metadata(self, days_back: Optional[int] = 7) -> List[TestMetadataSchema]:
+        tests_metadata = json.loads(self.dbt_runner.run_operation(macro_name='get_test_results', macro_args=dict(days_back=days_back))[0])
+        return [TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata]
+
+    def get_metrics(
+        self,
+        tests_metadata: Optional[List[TestMetadataSchema]] = None,
+        days_back: Optional[int] = None,
+        metrics_sample_limit: int = 5
+    ) -> Dict[TestUniqueIdType, Dict[str, Any]]:
+        tests: List[TestMetadataSchema] = tests_metadata if tests_metadata is not None else self.get_tests_metadata(days_back=days_back)
+        tests = [dict(test) for test in tests]
+        return json.loads(self.dbt_runner.run_operation(
+            macro_name='get_tests_metrics',
+            macro_args=dict(tests=tests, metrics_sample_limit=metrics_sample_limit)
+        )[0])
+
+    def get_invocations(
+        self,
+        invocations_per_test: int = 30
+    ) -> Dict[TestUniqueIdType, InvocationsSchems]:
+        tests_invocations = dict()
+        raw_invocations = json.loads(self.dbt_runner.run_operation(
+            macro_name='get_tests_invocations',
+            macro_args=dict(invocations_per_test=invocations_per_test)
+        )[0])
+        for sub_test_unique_id, invocations in raw_invocations.items():
+            test_invocations = []
+            test_invocations_times = json.loads(invocations["invocations_times"])
```

Comment:
I think that if you use array_agg you won't need this json load again

Reply:
Can't use array_agg because it is not supported by all platforms (as mentioned before)